kt/plugins/godot-gradle-plugin/src/main/kotlin/godot/gradle/tasks/initialKotlinCompileForClassGraph.kt

```diff
@@ -0,0 +1,49 @@
+package godot.gradle.tasks
+
+import godot.gradle.projectExt.kotlinJvmExtension
+import org.gradle.api.Project
+import org.gradle.api.tasks.SourceSetContainer
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.api.tasks.compile.JavaCompile
+import org.gradle.api.tasks.scala.ScalaCompile
+import org.jetbrains.kotlin.gradle.dsl.KotlinJvmCompilerOptions
+import org.jetbrains.kotlin.gradle.dsl.KotlinJvmOptions
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinWithJavaCompilation
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+fun Project.initialKotlinCompileForClassGraph(
+    classGraphScalaCompile: TaskProvider<ScalaCompile>,
+    classGraphJavaCompile: TaskProvider<JavaCompile>
+): KotlinWithJavaCompilation<KotlinJvmOptions, KotlinJvmCompilerOptions> {
```

Comment:
KotlinJvmOptions is deprecated

Reply:
After examination i think we can ignore this deprecation. They still use it internally and it's what we get from the kotlin plugin when creating a compilation.
I think it's deprecated beacause it was initially exposed for configuration by end users and now that configuration is done differently. But i think we can and should still rely on it internally.